File: src/php/Generic/IfElseBenchmark.php
<?php

namespace EntityFX\NetBenchmark\Core\Generic {

    use EntityFX\NetBenchmark\Core\Writer;
    
    class IfElseBenchmark extends BenchmarkBase {

        public function __construct($writer, $printToConsole)
        {
            parent::__construct($writer, $printToConsole);
            $this->Iterrations = 2000000000;
            $this->Ratio = 0.01;
        }


        protected function BenchImplementation()
        {
            $d = 0;

            for ($i = 0, $c = -1; $i < $this->Iterrations; $i++, $c--)
            {
                $c = $c === -4 ? -1 : $c;
                if ($i === -1)
                {
                    $d = 3;
                }
                else if ($i === -2)
                {
                    $d = 2;
                }
                else if ($i === -3)
                {
                    $d = 1;
                }
                $d = $d + 1;
            }
            return $d;
        }
    }
}

File: src/lua/generic/ifElseBenchmark.lua
IfElseBenchmark = class(BenchmarkBase, function(a, writer, printToConsole)
    BenchmarkBase.init(a, writer, printToConsole)
    a.name = "IfElseBenchmark"
    a.iterrations = 2000000000
    a.ratio = 0.01
end)

function IfElseBenchmark:benchImplementation()
    local d = 0
    local c = -1
    for i=1,self.iterrations do
        c = (c == -4 and -1 or c)
        if i == -1 then
            d = 3
        elseif i == -2 then
            d = 2
        elseif i == -3 then
            d = 1
        end
        d = d + 1
        c = c - 1
    end
    return d
end